### PR TITLE
feat: improvements to portal

### PR DESCRIPTION
**Pull request by satya164**
- layer management according to position
- preserve theme context via ThemedPortal

### CODE REVIEW

File: example/main.js [hunk 1]
```diff
@@ -11,7 +11,11 @@ import {
   NavigationProvider,
   StackNavigation,
 } from '@exponent/ex-navigation';
-import { Colors, ThemeProvider, Drawer } from 'react-native-paper';
+import {
+  Colors,
+  Drawer,
+  Provider as PaperProvider,
```
Comment (ahmedlhanafy) [inline example/main.js]: Why not naming it `PaperProvider`?

File: src/index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 export { default as withTheme } from './core/withTheme';
 export { default as ThemeProvider } from './core/ThemeProvider';
-export { default as PortalHost } from './core/PortalHost';
+export { default as Provider } from './core/Provider';
```
Comment (ahmedlhanafy) [inline src/index.js]: Why didn't you export ThemedPortal?

File: src/components/Portal/Portal.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import {
   PureComponent,
   PropTypes,
 } from 'react';
-import { manager } from '../core/PortalHost';
+import { manager } from './PortalHost';
 
 type Props = {
```
Comment (ahmedlhanafy) [inline src/components/Portal/Portal.js]: Why didn't you add `position` prop here?

File: src/components/Portal/ThemedPortal.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  Children,
+  PropTypes,
+} from 'react';
+import Portal from './Portal';
+import ThemeProvider from '../../core/ThemeProvider';
+import withTheme from '../../core/withTheme';
+import type { Theme } from '../../types/Theme';
+
+type Props = {
+  children?: any;
```
Comment (ahmedlhanafy) [inline src/components/Portal/ThemedPortal.js]: What about adding a `position` prop here as well to inform the user, without assuming that he knows that this component uses portal and passes its props to the portal?

File: example/main.js (hunk 1 above)
Comment (satya164) [inline example/main.js]: I don't like the name very much (sounds like it provides a paper, lol). Just copied what Redux did. Tell me if you have a good name in mind.

File: src/index.js (hunk 2 above)
Comment (satya164) [inline src/index.js]: @ahmedlhanafy I want experiment around Portal a bit more before deciding what to export.

File: src/components/Portal/Portal.js (hunk 3 above)
Comment (satya164) [inline src/components/Portal/Portal.js]: Will add that

**Pull request merged**